upd readme (#1)
DevExpress-Examples/asp-net-web-forms-gridview-custom-buttons-in-batch-edit-mode: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/T114462)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

> # GridView for Web Forms - How to implement custom buttons in status bar in batch edit mode

Reviewer: ```suggestion
# GridView for Web Forms - How to implement custom buttons in the status bar in batch edit mode
```

<!-- run online -->
**[[Run Online]](https://codecentral.devexpress.com/t114462/)**
<!-- run online end -->

> In [batch edit mode](https://docs.devexpress.com/AspNet/16443/components/grid-view/concepts/edit-data/batch-edit-mode), the Grid displays three buttons in [status bar](https://docs.devexpress.com/AspNet/3693/components/grid-view/visual-elements/status-bar): _Preview changes_, _Save changes_, and _Cancel changes_. You can not change the buttons or their behavior. This example demonstrates how to replace the default buttons with custom buttons.

Reviewer: 1. In the first sentence, please add an article ("the") before "status bar".

2. `You can not change the buttons or their behavior. This example demonstrates how to replace the default buttons with custom buttons.`
a) In the initial sentence you tell the reader that they canot change the buttons, but in the sentence that follows you mention that they can be replaced with default buttons. Please clarify.
b) Please change "can not" to "cannot".

For example, do you mean the following?

`You cannot change the behavior of the default buttons, but you can replace them with custom buttons. See the example below for details.`

Author: Yes, your version is correct.
```suggestion
In [batch edit mode](https://docs.devexpress.com/AspNet/16443/components/grid-view/concepts/edit-data/batch-edit-mode), the Grid displays three buttons in the [status bar](https://docs.devexpress.com/AspNet/3693/components/grid-view/visual-elements/status-bar): _Preview changes_, _Save changes_, and _Cancel changes_. You cannot change the behavior of the default buttons, but you can replace them with custom buttons. See the example below for details.
```

![Grid View - Confirmation message](grid-custom-buttons.png)

To replace the default buttons with custom buttons, implement the [StatusBar](http://docs.devexpress.devx/AspNet/DevExpress.Web.GridViewTemplates.StatusBar) template.

```aspx
<Templates>
  <StatusBar>
    <dx:ASPxButton ClientInstanceName="btnPreview"... />
    <dx:ASPxButton ClientInstanceName="btnSave"... />
    <dx:ASPxButton ClientInstanceName="btnCancel"... />
  </StatusBar>
</Templates>
```

> Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons repeat the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text when clicked.

Reviewer: `In this example, custom buttons repeat the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text when clicked.`

1) What do you mean by "repeat"? For example, do you mean "imitate"?

2) `the _Show changes_ button changes its text when clicked`

Do we need to mention the _new_ text to which this text changes?

Author: ```suggestion
Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons imitate the default buttons' behavior: they are hidden when the Grid has no changes and the _Show changes_ button changes its text to _Hide preview_ when clicked.
```

Reviewer: Thank you. I'd also like to avoid the plural possessive noun. Can we say the following?

```suggestion
Handle the [BatchEditEndEditing](http://docs.devexpress.devx/AspNet/js-ASPxClientGridView.BatchEditEndEditing) event and change the button visibility based on your conditions. In this example, custom buttons imitate the behavior of default buttons: they are hidden when the Grid has no changes and the _Show changes_ button changes its text to _Hide preview_ when clicked.
```
